api/tests/routes/backoffice/pro_test.py

```diff
@@ -1680,3 +1681,275 @@ def test_connect_as_bank_account_without_eligible_user(self, authenticated_clien
+        redirected_response = authenticated_client.get(response.location)
+        assert (
+            html_parser.extract_alert(redirected_response.data)
+            == "Aucun utilisateur approprié n'a été trouvé pour se connecter à cette offre collective"
+        )
+
+    @override_features(WIP_CONNECT_AS_EXTENDED=True)
+    def test_connect_as_collective_offer_template(self, authenticated_client, legit_user):
+        user_offerer = offerers_factories.UserOffererFactory()
+        offer = educational_factories.CollectiveOfferTemplateFactory(
+            venue__managingOfferer=user_offerer.offerer,
+        )
+        form_data = {"object_type": "collective_offer_template", "object_id": offer.id, "redirect": "/venue"}
+        expected_token_data = {
+            "user_id": user_offerer.userId,
+            "internal_admin_id": legit_user.id,
+            "internal_admin_email": legit_user.email,
+            "redirect_link": settings.PRO_URL + "/venue",
+        }
+
+        response = self.post_to_endpoint(
+            authenticated_client,
+            form=form_data,
+            expected_num_queries=self.expected_num_queries_without_user,
+        )
+
+        assert response.status_code == 303
+        base_url, key_token = response.location.rsplit("/", 1)
+        assert base_url + "/" == urls.build_pc_pro_connect_as_link("")
+        assert SecureToken(token=key_token).data == expected_token_data
+
+    @override_features(WIP_CONNECT_AS_EXTENDED=False)
+    def test_connect_as_collective_offer_template_protected_by_feature_flag(self, authenticated_client):
+        offer = educational_factories.CollectiveOfferTemplateFactory()
+
+        form_data = {"object_type": "collective_offer_template", "object_id": offer.id, "redirect": "/venue"}
+        response = self.post_to_endpoint(
+            authenticated_client,
+            form=form_data,
+            expected_num_queries=self.expected_num_queries_without_user,
+        )
+
+        assert response.status_code == 303
+        assert response.location == url_for("backoffice_web.home", _external=True)
+        redirected_response = authenticated_client.get(response.location)
+        assert (
+            html_parser.extract_alert(redirected_response.data)
+            == "L'utilisation de la version étendue de « connect as » requiert l'activation de la feature : WIP_CONNECT_AS_EXTENDED"
+        )
+
+    @override_features(WIP_CONNECT_AS_EXTENDED=True)
+    def test_connect_as_collective_offer_template_without_user(self, authenticated_client):
+        offer = educational_factories.CollectiveOfferTemplateFactory()
+
+        form_data = {"object_type": "collective_offer_template", "object_id": offer.id, "redirect": "/venue"}
+        response = self.post_to_endpoint(
+            authenticated_client,
+            form=form_data,
+            expected_num_queries=self.expected_num_queries_without_user + 1,  # +1 for rollback query
```

Comment:
Question pour ma culture perso -> pourquoi a-t-on besoin de mettre à jour cette valeur (si on ne test pas le nombre de queries) ?

Reply:
en fait in test le nombre de queries au niveau du post (on peut pas utiliser le context manager parcequ'avant de faire un post on fait un get pour récuperer le csrf ce qui rajoute des requetes a la db)

Reply:
Nice. Je viens de lire le code de `post_to_endpoint`, c'est cool !